test/WebSites/InlineConstraintsWebSite/Startup.cs

```diff
@@ -12,6 +15,27 @@ public class Startup
         // Set up application services
         public void ConfigureServices(IServiceCollection services)
         {
+
+            services.Configure<RouteOptions>(
+                routeOptions => routeOptions.ConstraintMap.Add(
+                    "producttype",
+                    typeof(ItemTypeConstraint)));
+
+            services.Configure<RouteOptions>(
+                routeOptions => routeOptions.ConstraintMap.Add(
+                    "servicetype",
+                    typeof(ItemTypeConstraint)));
+
+            services.Configure<RouteOptions>(
+                routeOptions =>
+                {
+                    if (routeOptions.ConstraintMap.ContainsKey("servicetype"))
+                    {
+                        routeOptions.ConstraintMap["servicetype"] =
+                            typeof(ItemTypeUpperCaseConstraint);
+                    }
+                });
```

Comment:
huh?, if you want to test an update scenario you can update an already existing one, you don't have to add and then update a constraint, for example if you wanted to update `exists` to mean something else. 
And lets combine all these adds as one lambda.